scripts/ida_toggles.py

```diff
@@ -44,20 +44,65 @@ def add_toggle_data(self, dump_file_path):
+            if toggle_name is None and row['model'] == "WaffleUtilsWaffleflagcourseoverridemodel":
+                toggle_name = row['fields']['waffle_flag']
             # convert sql dump model name to annotation report toggle type name
             if row['model'] == "waffle.flag":
                 toggle_type = "WaffleFlag"
             elif row['model'] == "waffle.switch":
                 toggle_type = "WaffleSwitch"
+            elif row['model'] == "WaffleUtilsWaffleflagcourseoverridemodel":
+                toggle_type = "CourseWaffleFlag/CourseOverride"
             else:
                 toggle_type = row['model']
 
             toggle_data = row['fields']
             toggle_state = ToggleState(toggle_type, toggle_data)
             toggle = Toggle(toggle_name, toggle_state)
+            if toggle_type not in self.toggles.keys():
+                self.toggles[toggle_type] = {}
+
+            if toggle_type == "CourseWaffleFlag/CourseOverride":
+                if toggle.name not in self.toggles[toggle_type].keys():
+                    # if a "CourseWaffleFlag/CourseOverride" was found in data before its corresponding
+                    # CourseWaffleFlag, create an empty toggle to add course override data
+                    toggle_state = ToggleState("CourseWaffleFlag", {})
+                    toggle = Toggle(toggle_name, toggle_state)
+                    self.toggles["CourseWaffleFlag"][toggle_name] = toggle
+
+                # add course override data to toggle output
+                # assuming each course only has one override for unique toggle
+                # get dict with all course_overrides and add new course to it
+                course_overrides = self.toggles["CourseWaffleFlag"][toggle.name].state.get_datum(
+                                        "course_overrides",
+                                        cleaned=False,
+                                        )
+
+                if not course_overrides:
+                    course_overrides = {}
+
+                course_overrides[toggle_data["course_id"]] = toggle_data["override_choice"]
+                self.toggles["CourseWaffleFlag"][toggle.name].state.set_datum(
+                    "course_overrides",
+                    course_overrides,
+                    cleaned=False,
+                )
+
+                LOGGER.info(
+                    'Adding override choice for course {} to waffle flag {}'.format(toggle_data["course_id"], toggle.name)
+                )
+            elif toggle_type == "CourseWaffleFlag":
+                # in case, overrides data already added this toggle to data, make sure not to delete data
+                # this should only be relevant if annotations data is added before toggle state data is added
+                if toggle_name in self.toggles[toggle_type].keys():
+                    for k, v in toggle_data:
+                        self.toggles[toggle_type][toggle_name].state.set_datum(k, v, cleaned=False)
```

Comment:
I didn't look too closely at this.  Does `dict.update()` give you what you want?  If not, can you create a small appropriately named helper function, so this becomes more clear with less comments and less code in this function?

Reply:
As it is currently, the dicts in state that hold data are "private" variables(prefixed _) and I wanted to follow the practice of not directly interacting with private variables.  They probably could be made none private and there is an argument to be made the way data structure in this code is way too confusing. But that would require a lot of refracting and even with refracting, I suspect a level of complexity would remain.

Reply:
How do you feel about a little helper with an appropriate name and/or docstring that helps clarify and move this complexity out of the main function?

Reply:
I like your suggestion of adding extra line

Reply:
I took your suggestion and seperated things into seperate functions. Which definitely made this wayyy cleaner